### PR TITLE
BCDA-1596 Feature: Added db model for suppression data

### PR DESCRIPTION
**Pull request by tbellj**
Fixes [BCDA-1596](https://jira.cms.gov/browse/BCDA-1596)

Problem:
We need to have our db models in place before implementing the 1-800 medicare suppression ETL.

Proposed changes:
Added db model for 1-800 medicare suppression data.

Change Details:
bcda/models/models.go - gorm db model code
db/api.sql - db schema updates

Security Implications:
Yes, this code **will** handle files containing PII/PHI.
This change does **not** add new software dependencies
This change does **not** alter security controls or supporting software
This change does **not** introduce storage of new data
A security checklist **has** been completed for this change

Acceptance Validation:
Yes, all tests pass

Feedback Requested:
Any and all

### CODE REVIEW

File: bcda/models/models.go [hunk 1]
```diff
@@ -347,6 +348,19 @@ type CCLFBeneficiary struct {
 	BlueButtonID string `gorm:"type: text"`
 }
 
+type Suppression struct {
+	gorm.Model
+	HICN                string `gorm:"type:varchar(11);not null"`
+	SourceCode          string `json:"source_code"`
```
Comment (embh) [inline bcda/models/models.go]: Should there be a length on the source code field? It has length 5 in the ICD (not clear if it's fixed or variable though).

File: bcda/models/models.go [hunk 2]
```diff
@@ -347,6 +348,19 @@ type CCLFBeneficiary struct {
...
+	EffectiveDt         time.Time
+	PrefIndicator       string `gorm:"type:char(1)"`
+	SAMHSASourceCode    string `json:"samhsa_source_code"`
```
Comment (embh) [inline bcda/models/models.go]: Same question as above (length 5)

File: bcda/models/models.go (hunk 1 above)
Comment (tbellj) [reply bcda/models/models.go]: good eyes! Yep I saw the same thing but the ICD also gives 2 examples of valid values...kinda contradicting their own ICD. They show 2 examples: "1800" and "1-800" one being length 4 and one being length 5, so I just went with a varchar...but maybe there's a better way. thoughts?

Comment (embh) [reply bcda/models/models.go]: Aha. In that case, `varchar(5)` probably works: variable length, max 5.

Comment (tbellj) [reply bcda/models/models.go]: nice!

File: bcda/models/models.go [hunk 3]
```diff
@@ -347,6 +348,19 @@ type CCLFBeneficiary struct {
...
+type Suppression struct {
+	gorm.Model
+	HICN                string `gorm:"type:varchar(11);not null"`
+	SourceCode          string `gorm:"type:varchar(5)"`
+	EffectiveDt         time.Time
+	PrefIndicator       string `gorm:"type:char(1)"`
+	SAMHSASourceCode    string `gorm:"type:varchar(5)"`
+	SAMHSAEffectiveDt   time.Time
+	SAMHSAPrefIndicator string `gorm:"type:char(1)"`
+	ACOCMSID            string `gorm:"type:char(5)"`
```
Comment (embh) [inline bcda/models/models.go]: I just tried a GORM migration locally, and the column names all look correct except `ACOCMSID` (ends up as `acocms_id`). So we probably need to specify the name for that one.

Comment (tbellj) [reply bcda/models/models.go]: ooo nice, good check too. Will update

**Pull request merged**